src/lib.rs

```diff
@@ -25,6 +25,40 @@ use std::ops::{Deref, DerefMut};
 /// Determines how a memory map may be used. If the memory map is backed by a
 /// file, then the file must have permissions corresponding to the operations
 /// the protection level allows.
+///
+/// # Example
+///
+/// ```rust
+/// # use std::error::Error;
+/// use std::fs::File;
+/// use std::io::Write;
+/// use memmap::Protection;
+///
+/// # fn try_main() -> Result<(), Box<Error>> {
+/// let file = File::open("README.md")?;
+///
+/// // Initialize a mutable memory map with `ReadCopy` protection
+/// let mut mmap = unsafe { memmap::file(&file)
+///                             .protection(Protection::ReadCopy)
+///                             .map_mut()? };
+///
+/// // This is allowed
+/// (&mut mmap[..]).write(b"Hello")?;
+///
+/// // This is also allowed
+/// mmap.set_protection(Protection::ReadWrite)?;
+///
+/// // This is NOT allowed and will return an error,
+/// // since we are trying to make a `MmapMut` read-only.
+/// mmap.set_protection(Protection::Read)?;
```

Comment:
So, quick question: what is an allowable `set_protection`? Because apparently going from `ReadCopy` to `ReadWrite` gives an error:
`code: 87, message: "The parameter is incorrect"`

But I can't put in a `ReadWrite` to `ReadCopy` example because it doesn't have permissions to open `README.md`. (Unless the doc test is `no_run`, but that defeats the purpose)

Also, as far as I can tell, there is no test for `set_protection`, so I don't have anything to work off of (and I don't really understand why a user would change the protection of a mutable mmap).

Thanks!